### PR TITLE
Create Gdl.de.xml

### PR DESCRIPTION
**Pull request by gbnrileuhg1**
My first pull request ever, please let me know if you have some hints :-)

### CODE REVIEW

**Comment by Bisaloo** [comment]
Thank you for your pull request!

This domain has a lot of other subdomains that you can find with `Sublist3r` (see [this relevant section](https://github.com/EFForg/https-everywhere/blob/master/CONTRIBUTING.md#enumerating-subdomains) in our contributing guide). Would you be interested in adding them as well?

**Comment by gbnrileuhg1** [comment]
While creating it I thought "nah, I'm lazy and this doesn't have much practical use". But while I did some other stuff in the meantime, that changed to "wait, that's actually not that hard and does have an effect". So yeah, I'll give it a try :-)

**Comment by jeremyn** [comment]
@gbnrileuhg1 This is a complex pull request to try for your first one. You might want to start with less complicated domains until you get the hang of proper formatting on so on, to avoid less back-and-forth discussion for the larger pull requests like this one.

**Comment by Bisaloo** [comment]
@gbnrileuhg1, I agree with @jeremyn here. In your list, you had some domains that may be a better for a first PR.

However, if you prefer to keep going here, please let me know.

**Comment by J0WI** [comment]
@gbnrileuhg1 would you like to finish this PR?

File: src/chrome/content/rules/Gdl.de.xml [hunk 1]
```diff
@@ -125,9 +125,9 @@
...
 	<securecookie host=".+" name=".+" />
 	
-	<rule from="^http://www.bezirk-(\l+)\.gdl\.de" to="https://bezirk-$1\.gdl\.de" />
-	<rule from="^http://www.og-(\l+)\.gdl\.de" to="https://og-$1\.gdl\.de" />
-	<rule from="^http://www.jugend\.gdl\.de" to="https://jugend\.gdl\.de" />
-	<rule from="^http://www.m\.gdl\.de" to="https://m\.gdl\.de" />
+	<rule from="^http://www.bezirk-(\l+)\.gdl\.de" to="https://bezirk-$1.gdl.de" />
+	<rule from="^http://www.og-(\l+)\.gdl\.de" to="https://og-$1.gdl.de" />
+	<rule from="^http://www.jugend\.gdl\.de" to="https://jugend.gdl.de" />
+	<rule from="^http://www.m\.gdl\.de" to="https://m.gdl.de" />
```
Comment (J0WI) [inline src/chrome/content/rules/Gdl.de.xml]: We also require trailing slashes.

Comment (pipboy96) [reply src/chrome/content/rules/Gdl.de.xml]: @J0WI thx

**Comment by pipboy96** [comment]
@J0WI I'm still not confident merging this, can you look for possible errors?

File: src/chrome/content/rules/Gdl.de.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	<target host="og-saarbrueckenpv.gdl.de" />
+	<target host="og-stuttgart.gdl.de" />
+	<target host="og-tuebingen.gdl.de" />
+	<target host="og-ulm.gdl.de" />
+	<target host="og-villingen.gdl.de" />
+
+	<securecookie host=".+" name=".+" />
+
+	<rule from="^http://www.jugend\.gdl\.de/" to="https://jugend.gdl.de/" />
+	<rule from="^http://www.m\.gdl\.de/" to="https://m.gdl.de/" />
```
Comment (J0WI) [inline src/chrome/content/rules/Gdl.de.xml]: The dots in `www.` need to be escaped.

Comment (pipboy96) [reply src/chrome/content/rules/Gdl.de.xml]: thanks for pointing out

**Comment by pipboy96** [comment]
Closing for now, this ruleset has many second level targets, which should be submitted as separate PRs if possible.

**Comment by J0WI** [comment]
This was already nearly finished. I removed the remaining obsolete targets.

**Comment by pipboy96** [comment]
@J0WI I think we should split this as separate files, even if we keep it as one PR.

**Comment by J0WI** [comment]
What would you split?

**Comment by pipboy96** [comment]
@J0WI 
```
	<target host="gdl-og-gera.de" />
	<target host="www.gdl-og-gera.de" />

<!-- cut here -->

	<target host="gdl-hof.de" />
	<target host="www.gdl-hof.de" />

<!-- cut here -->

	<target host="gdl-kempten.de" />
	<target host="www.gdl-kempten.de" />

<!-- cut here -->

	<target host="gdl-sbahn-berlin.de" />
	<target host="www.gdl-sbahn-berlin.de" />
```

**Comment by J0WI** [comment]
@Bisaloo are you fine with this?

File: src/chrome/content/rules/Gdl.de.xml [hunk 3]
```diff
@@ -0,0 +1,160 @@
+<!--
+	Other GDL rulesets:
+		- Gdl-hof.de.xml
+		- Gdl-kempten.de.xml
+		- Gdl-og-gera.de.xml 
+		- Gdl-sbahn-berlin.de.xml
+
+	Valid, but redirects to plain HTTP:
+		- gdl-og-mannheim.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: I can't resolve this domain

File: src/chrome/content/rules/Gdl.de.xml [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+	Server-side SSL Error:
+		- gdl-aachen.de
+		- gdl-bayern.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: SSL works but MCB

File: src/chrome/content/rules/Gdl.de.xml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+		- gdl-friedrichshafen.de
+		- gdl.koeln (geoTLD)
+		- gdl-mühldorf.de (xn- -gdl-mhldorf-xhb.de)
+		- gdl-nn.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: WFM

File: src/chrome/content/rules/Gdl.de.xml [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+		- gdl-tuebingen.de
+
+	Mismatched:
+		- umfrage.gdl.de
+		- gdl-bremen.de
+		- gdl-garmisch.de
+		- gdl-heilbronn.de
+		- gdl-kassel.de
+		- gdl-kornwestheim.de
+		- gdl-m.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: WFM as well as https://www.gdl-m.de/

File: src/chrome/content/rules/Gdl.de.xml [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+		- gdl-regensburg.de
+		- gdl-ulm.de
+		- gdl-weiden-schwandorf.de
+		- gdl-wuerzburg.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: ERR_SSL_UNRECOGNIZED_NAME_ALERT

File: src/chrome/content/rules/Gdl.de.xml [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+	Certificate Error Unknown Issuer:
+		- sentry.gdl.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: gone from DNS

File: src/chrome/content/rules/Gdl.de.xml [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+		
+	Plaintext response on HTTPS:
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: What does this mean?

File: src/chrome/content/rules/Gdl.de.xml [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+		- www.jugend.gdl.de
+
+ 	Other Connection Issue:
+		- gdl-og-dresden.de
+		- gdlhamm.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: WFM

File: src/chrome/content/rules/Gdl.de.xml [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+		- gdl-ludwigshafen.de
+		- gdl-nord.de
+		- gdl-rothensee.de
```
Comment (Bisaloo) [inline src/chrome/content/rules/Gdl.de.xml]: Domain for sale

File: src/chrome/content/rules/Gdl.de.xml (hunk 9 above)
Comment (pipboy96) [reply src/chrome/content/rules/Gdl.de.xml]: @Bisaloo It effectively means a plaintext HTTP server is listening on port 443 due to misconfiguration.

File: src/chrome/content/rules/Gdl.de.xml (hunk 7 above)
Comment (J0WI) [reply src/chrome/content/rules/Gdl.de.xml]: Got a mismatch in browser.

**Comment by Bisaloo** [comment]
LGTM, should I merge now?

**Comment by J0WI** [comment]
Sure!

**Pull request merged**